blocks/note/note.js

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright 2022 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+export default function decorate(block) {
```

Comment:
I'm a little confused about who calls `decorate`. Is it supposed to be called somewhere in scripts.js? 
It makes sense that milo components are handled by milo (through `loadArea` and `loadDelayed` which decorate the sections and load each individual block using `loadBlock`, which calls the block's `init` function) and our components are handled by us, but how do we override milo components (should we need to) while avoiding wasted work?

Reply:
This is a Franklin/Milo convention where, when you load a page, it will look at `.section` divs and for each of them, it will look at the extra class which is the block name, here, `note` (so the element `<div class="section note"></div>`).

Then, in Milo project it will try to get the `note` block (download css+js) from 2 ways:
1. From the existing Milo blocks
2. If not found in Milo blocks, it will look for `/blocks/note/note.js` and `/blocks/note/note.css`
Then, having the code+css, from the `note.js` file, it will execute the default exported function (here `decorate` but it could be another name).

That's why the function is not explicitely called anywhere in the code.
I suggest you look at official documentation page https://www.hlx.live/developer/markup-sections-blocks.

Also, do not confuse Milo blocks and custom blocks with (what I called) global blocks:
1. Milo blocks, a library of blocks handled by Milo team. The code of these blocks is not in customer project, the code is in https://github.com/adobecom/milo. How do your Milo project get these blocks? https://github.com/adobecom/milo-college, the template project from which you create your Milo project, contains a bit of extra javascript code to get the library loaded in every page
2. Custom blocks: some of your pages need a new block which is not in Milo, and to add it to your page, you need the author to add some content in the document (like the note in our case). Then, you add a block in your project, in folder `/blocks/<block_name>`. Milo projects are configured to get the pages know how to get the block (code +css) when needed
3. Global blocks: ALL your pages have the same component (here, the internal banner), so you do not want to add something to every document. Then, you leverage the existing `/scripts/scripts.js` and `/styles/styles.css` to add the code and css for adding a component on every page.

Reply:
Thanks. This was super helpful in understanding Milo. It still seems like there's no easy way to overload a Milo block (since if a block exists in Milo, Milo will load it's own block and not look in our blocks folder).